File: Grokking Algorithms/quicksort.py
def quicksort(arry):
    if len(arry) < 2:
        return arry
    else:
        pivot = arry[0]  # 基准值
        left = []  # 小于基准值的元素
        right = []  # 大于基准值的元素
        for item in arry[1:]:
            if item > pivot:
                left.append(item)
            else:
                right.append(item)
        return quicksort(right) + [pivot] + quicksort(left)


print(quicksort([1, 3, 2, 8, 7, 5, 6]))

File: AhaAlgorithms/quickSort.cpp
int a[101];

void quicksort(int left, int right)
{
    if (left > right)
    {
        return;
    }
    int temp = a[left];
    int i = left;
    int j = right;
    int t;
    while (i != j)
    {
        while (a[j] >= temp && i < j)
        {
            --j;
        }
        while (a[i] <= temp && i < j)
        {
            ++i;
        }
        if (i < j)
        {
            t = a[i];
            a[i] = a[j];
            a[j] = t;
        }
    }
    a[left] = a[i];
    a[i] = temp;
    quicksort(left, i - 1);
    quicksort(i + 1, right);
}

int main()
{
    return 0;
}